### PR TITLE
Repair ruby 3.1.x compability with backwards compability. Bumped vers…

### LINKED ISSUE #325: Incompatability with ruby 3.1.0
I tried to install hiera-eyaml on my M1 Mac and got the following error using ruby 3.1.0:
```
undefined method `dependencies' for #<Gem::StubSpecification:0x0000000104eb38e0 @extension_dir=nil, @full_gem_path=nil, @gem_dir=nil, @ignored=nil, @loaded_from="/Users/m214042/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/specifications/CFPropertyList-3.0.5.gemspec", @data=#<Gem::StubSpecification::StubLine:0x0000000104eb1838 @name="CFPropertyList", @version=#<Gem::Version "3.0.5">, @platform="ruby", @extensions=[], @full_name="CFPropertyList-3.0.5", @require_paths=["lib"]>, @name=nil, @spec=nil, @base_dir="/Users/m214042/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0", @gems_dir="/Users/m214042/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems", @default_gem=false> (NoMethodError)

              dependency = spec.dependencies.find { |d| d.name == "hiera-eyaml" }
                               ^^^^^^^^^^^^^
```
Switching back to ruby 2.7.5. resolved the issue for me.

### PR DESCRIPTION
**Pull request by mmachner**
should fix #325 and #327

### CODE REVIEW

File: lib/hiera/backend/eyaml.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@ class Hiera
   module Backend
     module Eyaml
 
-      VERSION = "3.2.2"
+      VERSION = "3.2.3"
```
Comment (bastelfreak) [inline lib/hiera/backend/eyaml.rb:5]: please do not bump the version. we will do that during the release process

Comment (mmachner) [reply lib/hiera/backend/eyaml.rb]: ok

File: lib/hiera/backend/eyaml/plugins.rb [hunk 2]
```diff
@@ -32,8 +32,8 @@ def self.find
           [index].flatten.each do |source|
             specs = gem_version >= Gem::Version.new("1.6.0") ? source.latest_specs(true) : source.latest_specs
 
-            specs.each do |stub|
-              spec = gem_version >= Gem::Version.new("3.3.0") ? stub.to_spec : stub
+            specs.each do |spec|
+              spec = spec.to_spec if spec.respond_to?(:to_spec)
```
Comment (bastelfreak) [inline lib/hiera/backend/eyaml/plugins.rb:36]: would be cool if we could test this somehow 🧐

Comment (mmachner) [reply lib/hiera/backend/eyaml/plugins.rb:36]: added 3.1 to the actions pipeline

Comment (bastelfreak) [reply lib/hiera/backend/eyaml/plugins.rb:36]: mhm I tried that in https://github.com/voxpupuli/hiera-eyaml/pull/326/commits and see aome odd errors there. could you add ruby 3.0 as well?

**Comment by genebean** [comment]
I cloned this locally, ran `gem build hiera-eyaml.gemspec` followed by `gem install hiera-eyaml-3.2.3.gem` and now have a working eyaml command on Fedora 36. Sample output below:

```bash
╔ ☕️  carbonbean:~/repos/hiera-eyaml (mmachner-bugfix/325-regression ✔)    
╚ᐅ gem build hiera-eyaml.gemspec 
WARNING:  open-ended dependency on optimist (>= 0) is not recommended
  use a bounded requirement, such as '~> x.y'
WARNING:  open-ended dependency on highline (>= 0) is not recommended
  use a bounded requirement, such as '~> x.y'
WARNING:  See https://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: hiera-eyaml
  Version: 3.2.3
  File: hiera-eyaml-3.2.3.gem

╔ ☕️  carbonbean:~/repos/hiera-eyaml (mmachner-bugfix/325-regression ✘) ✭   
╚ᐅ gem install hiera-eyaml-3.2.3.gem 
WARNING:  You don't have /home/gene/bin in your PATH,
	  gem executables will not run.
Successfully installed hiera-eyaml-3.2.3
Parsing documentation for hiera-eyaml-3.2.3
Installing ri documentation for hiera-eyaml-3.2.3
Done installing documentation for hiera-eyaml after 0 seconds
1 gem installed

╔ ☕️  carbonbean:~/repos/hiera-eyaml (mmachner-bugfix/325-regression ✘) ✭   
╚ᐅ which eyaml                     
~/bin/eyaml

╔ ☕️  carbonbean:~/repos/hiera-eyaml (mmachner-bugfix/325-regression ✘) ✭   
╚ᐅ eyaml --help
Welcome to eyaml 3.2.3 

Usage:
eyaml subcommand [global-opts] [subcommand-opts]

Available subcommands:
        encrypt: encrypt some data                                                
        recrypt: recrypt an eyaml file                                            
        version: show version information                                         
           edit: edit an eyaml file                                               
     createkeys: create a set of keys with which to encrypt/decrypt eyaml data    
        decrypt: decrypt some data                                                

For more help on an individual command, use --help on that command

Installed Plugins:
```

File: lib/hiera/backend/eyaml.rb (hunk 1 above)
Comment (bastelfreak) [reply lib/hiera/backend/eyaml.rb]: I cleaned this up and did a rebase

**Pull request merged**